includes/class-newspack-popups.php

```diff
@@ -41,6 +43,7 @@ public static function instance() {
 	 * Constructor.
 	 */
 	public function __construct() {
+		add_action( 'init', [ __CLASS__, 'create_lightweight_api_config' ] );
```

Comment:
I'd be concerned if that's fired on plugin update 🤔 Also following the advice [here](https://core.trac.wordpress.org/ticket/14170#comment:68), I've modified it so there's just one chance to create the config file: 3816b28e252c632b294de7ee9dc814a01bb699b0
Is that ok or too restrictive?

Reply:
That looks like a nice solution. 👍 I was thinking further, and since the activation hook wouldn't fire on plugin update we'd have had to deactivate then reactivate the plugin on all of our sites if we went with my suggestion. This latest approach should work nicely, and we won't have to do anything extra after deploying it. Good idea!